### PR TITLE
mysql-compatibility: Improve MySQL Compatibility

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

This improves some aspects of describing MySQL compatibility.

**security issues (such as `SELECT INTO OUTFILE`)** is not translated according to https://github.com/pingcap/docs/pull/3192/files#diff-7028919502ad5a35697a652366fdafadR12.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3151
- Other reference link(s):

### CODE REVIEW

File: mysql-compatibility.md [hunk 1]
```diff
@@ -8,7 +8,11 @@ aliases: ['/docs-cn/dev/reference/mysql-compatibility/']
 
 - TiDB 100% 兼容 MySQL 5.7 协议、MySQL 5.7 常用的功能及语法。MySQL 5.7 生态中的系统工具（PHPMyAdmin、Navicat、MySQL Workbench、mysqldump、Mydumper/Myloader）、客户端等均用于 TiDB。
 
-- 由于 TiDB 是一款分布式数据库，MySQL 5.7 中的部分特性因工程实现难度较大，投入产出比较低等多种原因在 TiDB 中未能实现或者仅兼容语法但功能并没有实现，因此使用过程中请特别注意。例如：`CREATE TABLE` 语句中 `ENGINE`，仅兼容语法，功能并没有实现，因此 TiDB 中没有 `ENGINE` 这类的概念。
+- 但 TiDB 尚未支持一些 MySQL 功能，可能的原因如下：
+    - 有更好的解决方案，例如 JSON 取代 XML 函数。
+    - 存在安全问题，例如 `SELECT INTO OUTFILE`。
```
Comment (TomShawn) [inline mysql-compatibility.md:13]:
```suggestion
```

File: sql-statements/sql-statement-analyze-table.md [hunk 2]
```diff
@@ -107,6 +107,11 @@ EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 
 * 在 MySQL 上不支持 `ANALYZE INCREMENTAL TABLE` 语句，它的使用可参考[增量收集文档](/statistics.md#增量收集)。
 
+TiDB 与 MySQL 的不同之处在于所收集的统计信息，以及查询执行过程中统计信息是如何被使用的。虽然 TiDB 中的 `ANALYZE` 语句在语法上与 MySQL 类似，但存在以下差异：
+
++ 执行 `ANALYZE TABLE` 时，TiDB 可能不包含最近提交的更改。对行进行批量更改后，在执行 `ANALYZE TABLE` 之前你可能需要先执行 `sleep(1)`，这样统计信息更新才能反映这些更改。见 [#16570](https://github.com/pingcap/tidb/issues/16570)。
```
Comment (TomShawn) [inline sql-statements/sql-statement-analyze-table.md:112]:
```suggestion
+ 执行 `ANALYZE TABLE` 时，TiDB 可能不包含最近提交的更改。若对行进行了批量更改，在执行 `ANALYZE TABLE` 之前你可能需要先执行 `sleep(1)`，这样统计信息更新才能反映这些更改。见 [#16570](https://github.com/pingcap/tidb/issues/16570)。
```

File: mysql-compatibility.md [hunk 3]
```diff
@@ -147,7 +124,7 @@ mysql> select _tidb_rowid, id from t;
 
 - `ONLY_FULL_GROUP_BY` 与 MySQL 5.7 相比有细微的[语义差别](/functions-and-operators/aggregate-group-by-functions.md#与-mysql-的区别)。
 
-- `NO_DIR_IN_CREATE` 和 `NO_ENGINE_SUBSTITUTION` MySQL 用于解决兼容问题，并不适用于 TiDB。
+- `NO_DIR_IN_CREATE` 和 `NO_ENGINE_SUBSTITUTION` MySQL 可用于解决兼容问题，并不适用于 TiDB。
```
Comment (lilin90) [inline mysql-compatibility.md:127]:
```suggestion
- `NO_DIR_IN_CREATE` 和 `NO_ENGINE_SUBSTITUTION` 仅用于解决与 MySQL 的兼容性问题，并不适用于 TiDB。
```

File: sql-statements/sql-statement-analyze-table.md [hunk 4]
```diff
@@ -107,6 +107,11 @@ EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 
 * 在 MySQL 上不支持 `ANALYZE INCREMENTAL TABLE` 语句，它的使用可参考[增量收集文档](/statistics.md#增量收集)。
 
+TiDB 与 MySQL 的不同之处在于所收集的统计信息，以及查询执行过程中统计信息是如何被使用的。虽然 TiDB 中的 `ANALYZE` 语句在语法上与 MySQL 类似，但存在以下差异：
```
Comment (lilin90) [inline sql-statements/sql-statement-analyze-table.md:110]:
```suggestion
TiDB 与 MySQL 在以下方面存在区别：所收集的统计信息，以及查询执行过程中统计信息是如何被使用的。虽然 TiDB 中的 `ANALYZE` 语句在语法上与 MySQL 类似，但存在以下差异：
```

File: sql-statements/sql-statement-analyze-table.md [hunk 5]
```diff
@@ -107,6 +107,11 @@ EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
...
+
++ 执行 `ANALYZE TABLE` 时，TiDB 可能不包含最近提交的更改。若对行进行了批量更改，在执行 `ANALYZE TABLE` 之前你可能需要先执行 `sleep(1)`，这样统计信息更新才能反映这些更改。见 [#16570](https://github.com/pingcap/tidb/issues/16570)。
```
Comment (lilin90) [inline sql-statements/sql-statement-analyze-table.md:112]:
```suggestion
+ 执行 `ANALYZE TABLE` 时，TiDB 可能不包含最近提交的更改。若对行进行了批量更改，在执行 `ANALYZE TABLE` 之前，你可能需要先执行 `sleep(1)`，这样统计信息更新才能反映这些更改。参见 [#16570](https://github.com/pingcap/tidb/issues/16570)。
```

**Comment by TomShawn** [comment]
@lilin90 Comments are addressed, PTAL again, thanks!

**Pull request merged**